### DISCUSSION TITLE
Filter earthaccess data with cloud coverage

### DISCUSSION THREAD

**Discussion by chabourque** [earthaccess-dev/earthaccess#445] [Q&A] [answered]
Hello,
I am using earthaccess.search_data to access HLS L30 & S30 data. I am working with a large area over a large period of time, meaning lots of data. I'd like to further filter my search to remove images with 30% cloud coverage to reduce the images I have to download. How can I do that?

**Comment by mfisher87** [comment, marked as answer]
You can do this with the `cloud_cover` argument to `search_data()`: 

```python
earthaccess.search_data(short_name="HLSS30",cloud_cover=(0,30))
```

https://earthaccess.readthedocs.io/en/latest/user-reference/granules/granules-query/#earthaccess.search.DataGranules.cloud_cover

**Reply by mfisher87** [reply]
@chabourque did this work for you? If so, can you select "Mark as answer"; otherwise, is there anything else we can do to support your work?

**Reply by chabourque** [reply]
Yes, it worked, thank you!

**Reply by mfisher87** [reply]
Amazing! :partying_face: Thanks for your post :)

**Reply by Kongstad** [reply]
Thanks for this! I was struggling to find this answer anywhere else and was close to conclude there was no filter option. 
I would kindly suggest that you put it in the python tutorial example!

Thank you

**Reply by mfisher87** [reply]
Any time. Thank you for the suggestion! I opened a ticket to document that work here: https://github.com/nsidc/earthaccess/issues/474

If you have any more ideas on how we can improve this documentation, we'd love to hear them over in that ticket :)
